In [4]:
import twitter

In [17]:
api = twitter.Api(consumer_key='WfQELhZV1qiIvYVfBc5wqO5xr',
                      consumer_secret='wdldaj2Hw6h7nRp2KTYYOLk7kkmkPA6SfXdwAppYoIQa5Z19P1',
                      access_token_key='1174300508384911365-U8PxqaeUtfyy7jV4Rwtc9DTU2IQj7L',
                      access_token_secret='zIoV392S1t062tFprvGhWwkqtCla5iMSRJdzYZa8VO7at')

In [100]:
results = api.GetSearch(
    raw_query="q=Bitcoin&result_type=mixed&since=2019-09-02&count=10000&lang=fr&tweet_mode=extended&max_id=1286441619499296500&since_id=1176441612492296192")
test = results
test

[Status(ID=1176503391230595076, ScreenName=Bitlionaire, Created=Tue Sep 24 14:27:21 +0000 2019, Text="Grâce à la loi Pacte, dès 2020, plus de 25 000 points de vente accepteront les paiements en #cryptomonnaie. D'abord le #Bitcoin puis d'autres altcoins à venir, avec le système Easy2Play\nhttps://t.co/E5CA0ZGWgA"),
 Status(ID=1176503089337180162, ScreenName=BitcoinPlanet2, Created=Tue Sep 24 14:26:09 +0000 2019, Text='Bakkt Futures Launch est un buste et Bitcoin tombe en dessous de 10 000 $ »NullTX #bitcoin #ethereum #blockchain #bitcoinplanet2 #crypto #tech #news #finance https://t.co/1kTRYL9bVJ'),
 Status(ID=1176502159006949376, ScreenName=DocCedef, Created=Tue Sep 24 14:22:27 +0000 2019, Text="RT @Conso_INC: Les #cryptoactifs : qu'est-ce que c'est ? Dans quelle mesure peuvent-ils être utilisés comme moyen de paiement ?\nPeuvent-ils…"),
 Status(ID=1176501912964915207, ScreenName=lamariejeanne75, Created=Tue Sep 24 14:21:29 +0000 2019, Text='RT @DirectViewInc: $DIRV Surveillance https:

In [105]:
results = api.GetSearch(
    raw_query="q=Bitcoin&result_type=mixed&since=2019-09-02&count=10000&lang=en&tweet_mode=extended")
test = results
test

[Status(ID=1175938226554658824, ScreenName=Bakkt, Created=Mon Sep 23 01:01:35 +0000 2019, Text='We’re live!\n\nThe first Bakkt Bitcoin Futures trade was executed at 8:02pm ET at a price of $10,115'),
 Status(ID=1176316072988831744, ScreenName=ErikVoorhees, Created=Tue Sep 24 02:03:01 +0000 2019, Text='If you bought Bitcoin this summer at $12,000 because it was going up, and you’re selling it now at $9,700 because it’s going down, you’re doing it wrong. “Hodl” isn’t just a cute meme... it’s wisdom from the elders who have been here before.'),
 Status(ID=1175945114671337472, ScreenName=Bakkt, Created=Mon Sep 23 01:28:57 +0000 2019, Text='Bakkt Bitcoin Futures are now actively trading on ICE Futures U.S., while the Bakkt Warehouse continues to onboard bitcoin\n\nWe take our customers’ trust seriously, and are excited to serve this market'),
 Status(ID=1176503985500512256, ScreenName=BitcoinDood, Created=Tue Sep 24 14:29:43 +0000 2019, Text="Government opposition to Libra is exactly what h

In [68]:
results = api.GetSearch(
    raw_query="q=Bitcoin&result_type=popular&since=2019-09-23&count=10000&lang=fr")
test = results
test

[Status(ID=1176441612492296193, ScreenName=Elsa_Trujillo_, Created=Tue Sep 24 10:21:52 +0000 2019, Text='Ce matin, plusieurs articles titraient que 25.000 commerces français accepteraient le bitcoin dès 2020.\n\nEn réalité… https://t.co/mH7MQqNmss'),
 Status(ID=1176373405622439936, ScreenName=Bloch_R, Created=Tue Sep 24 05:50:50 +0000 2019, Text='@Global_POS @Decathlon @boulanger @footlocker @Conforama @Intersport_FR @NorautoFr @easywallet 5/ Pour convaincre l… https://t.co/zB4jrg1rBS'),
 Status(ID=1176372815450259456, ScreenName=Bloch_R, Created=Tue Sep 24 05:48:29 +0000 2019, Text='@Global_POS @Decathlon @boulanger @footlocker @Conforama @Intersport_FR @NorautoFr 4/ Concrètement, un client d’une… https://t.co/iphRlRI7mz'),
 Status(ID=1176374718640902145, ScreenName=Bloch_R, Created=Tue Sep 24 05:56:03 +0000 2019, Text='@Global_POS @Decathlon @boulanger @footlocker @Conforama @Intersport_FR @NorautoFr @easywallet 6/ Rien n’est encore… https://t.co/tNZ0UkdR98')]

In [18]:
previous_Id = 1176441612492296192
result_Text = []
for i in range(1,149) :
    next_Id= previous_Id + 300000000000000000
    results = api.GetSearch(
        raw_query=f"q=Bitcoin&result_type=mixed&since=2019-09-02&count=10000&lang=en&tweet_mode=extended&max_id={next_Id}&since_id={previous_Id}")
    for j in results :
        result_Text.append(j.full_text)
    previous_Id = next_Id


In [19]:
file = open('test3.txt',"w", encoding='utf-8')
for text in result_Text :
    print(text)
    text = text
    file.write(text)
file.close()

Just read a magazine that fucked up my day
How you rate music that thugs with nothin' relate to it?
I help them see they way through it—not you
Can't step in my pants, can't walk in my shoes
Bet everything you worth, you'll lose your tie and your shirt
#Bitcoin DUMP IT! 🐻📉📉
RT @APompliano: The more people understand how the financial system is designed to control them and keep them down, the more people are dra…
RT @1MarkMoss: One of the early YouTube stars quit, and is coming back excited about Bitcoin!

“I’m actually more excited about Bitcoin tha…
Entries completed for the #Litecoin @TheWaterProject 50/50. Join me in about an hour for our live lotto drawing. Winner gets about 2.6 #LTC and the other will be donated. Thanks to everyone who bought tickets. #Litecoin #LTC #LitecoinFam #Bitcoin #BTC #Crypto
Despite the controversies, Tether has become the most traded crypto and has ~18% higher monthly trading volume than that of Bitcoin, according to CoinMarketCap (Olga Kharif/Bloomberg)

@CitizenBitcoin @giacomozucco Does bitcoin ensure a socialist free future, or can it be co-opted or corrupted by socialism somehow?
Crypto Weekly Highlights | Seized Bitcoin Auctioned

https://t.co/SnFcPSAkkb
Report: Wash Trading Plunges by 35% as Crypto Exchanges Start to Play Fair - https://t.co/WrnudIl4Ep
 #bitcoin #btc #bitcoinnews #blockchain https://t.co/m5ZxEoeqZO
RT @JuicyCSG: $500 BITCOIN GIVEAWAY! 

FOLLOW - @JuicyCSG 
FOLLOW - @BitcasinoIO 
SIGN UP AT (optional) - https://t.co/QT4TmDZVTT
RETWEET A…
#Crypto #cryptocurrency #bitcoin #btc #eth #ethereum #ltc #litecoin #xrp #ripple.
The race to $1 is on
$btt $xem
$doge $bat $vet $vsys $dgb $bts $bcn $sc $btm $iost $ardr $nexo $aoa 
$cnn 
$npxs
$xvg
$xrp
$hot
$qkc
$wtc
$pcl
IMO, PCL is going to reach $1 first.
Research why.
RT @WhaleTrades: BitMEX $BTC Whale:
$2,000,000 worth of #Bitcoin bought at $8,260.95 13:23:48 2019/10/02
| 💰💰💰💰💰 "Hey Hey Heyyyyyyyyy"
RT @DiepSanh: - How Catherine Coley, CEO of Binance US, gained the experie

RT @CCNMarkets: Experts See Bitcoin Rallying to $20,000 Before End of the Year; Here are the Reasons https://t.co/WpMxeN7lqa
Op Ed: Bitcoin and the Dawn of the Negative Interest Rate Era - https://t.co/bShJjoddiX
 #bitcoin #btc #bitcoinnews #blockchain https://t.co/NgcZ4vb0fT
This Germany State Made Passive Income via Bitcoin https://t.co/HcK1lGEioy
RT @APompliano: The more people understand how the financial system is designed to control them and keep them down, the more people are dra…
This Germany State Made Passive Income via Bitcoin https://t.co/cAkzt9spyD
RT @Airdropnotecom: #Bitcoin #Satoshi #crypto #blockchain #Airdrop
New Airdrop #Blackclass 📢

BLC airdrops 100,000 #BLC (~$20) to 10,000 me…
If you relax a little, I'm glad I inspire you, but Stan
Why are you so mad? Try to understand
That I do want you as a fan
I just don't want you to do some crazy shit
Zavvi: Champions League final giveaway ad breached rules 
BBCWorld - Twitter - News - Noticias - Bitcoin - CryptoCurrency - @

bitcoin is dreadful ☄️
Op Ed: Belarus and the Case for National Bitcoin Strategies - https://t.co/eEcn63zamC
 #bitcoin #btc #bitcoinnews #blockchain https://t.co/bf6JM1rHqa
New post (Bitcoin’s Scarcity Will Trigger Its Price to Hit $1M in 2020) has been published on UserInterface | UIX Network - https://t.co/YBslQoENZU https://t.co/7M96LuGbso
This Germany State Made Passive Income via Bitcoin https://t.co/5DZ4tg0SO0
(Bitcoin’s Scarcity Will Trigger Its Price to Hit $1M in 2020) - https://t.co/cNs8Graxwu #UIX https://t.co/RIn7YyvNYh
@suhasdaftuar We'll get started in about ten minutes. Lots to unpack this week. Feel free to join in on the #bitcoin-core-pr-reviews IRC channel.
@CitizenBitcoin @giacomozucco Does bitcoin ensure a socialist free future, or can it be co-opted or corrupted by socialism somehow?
Crypto Weekly Highlights | Seized Bitcoin Auctioned

https://t.co/SnFcPSAkkb
Report: Wash Trading Plunges by 35% as Crypto Exchanges Start to Play Fair - https://t.co/WrnudIl4Ep
 #bitc

Next block fee rate: 5 sat/vByte
Half hour fee rate: 2 sat/vByte
Hour fee rate: 2 sat/vByte

You can broadcast now your Bitcoin transaction 😎

#BroadcastNow
@PeterMcCormack @bitcoin_bolsa This is my eternal battle, what have you learned about winning this battle??
RT @cryptochaseio: Are we going to see Altseason any time soon? 2017 They were flourishing. This year they are not really the spitting imag…
RT @TamasBlummer: Bitcoin exists because there is demand for money with its properties. Technology just delivers what people want.
RT @BobLoukas: Will be working on a new Bitcoin #4yearjourney video today.  Any topics you want to hear about?
RT @DHillemann: My Podcast - where it all began

EP. 1 - The Big Why - What is #Blockchain - What is it not

Also available on #Anchor #App…
THE REVOLUTION WILL NOT BE TELEVISED... Day 2: CoinGeek Seoul Conference 2019 https://t.co/QbkRdbThqt #bitcoin #bsv #BitcoinSV
"Nuanced Analysis of LocalBitcoins Data Suggests Bitcoin is Working as Satoshi Inten

RT @CryptoFights: Check out some of the exclusive weapons only for CryptoFights! Be sure to follow us in the forums for a chance to win pri…
RT @BVBTC: "Nuanced Analysis of LocalBitcoins Data Suggests Bitcoin is Working as Satoshi Intended"

Thanks @MartyBent for writing about th…
START BITCOIN MINING       https://t.co/dPlJzmqpLA
RT @RoadtoRoota: I sat down with my old friend Sean at SGT Report to chat about the state of the world.
 
Here's the link:
 
This Feels Lik…
Russian nuclear scientist caught mining bitcoin in top-secret lab https://t.co/aS63s8Y4OH
RT @CoinCapsAi: #bitpay to #add #support for #xrp #cryptocurrency later this #year 
#instaico #instaairdrop #bitcoin #ico #ethereum #ripple…
RT @qhfofficial: Hey #Vietnam👋
Another publication is there🔥🔥🔥
This time made by https://t.co/bAFodhxQdw 

Check it out! 👇
https://t.co/MtL…
STILL Time to be a Bitcoin Maximalist? https://t.co/f2WleblfCF https://t.co/6Lsjop76tc
Bitcoin Jumps on Launch of Institutional ETF https://t.co/pjz1SBN6o

Despite the controversies, Tether has become the most traded crypto and has ~18% higher monthly trading volume than that of Bitcoin, according to CoinMarketCap (Olga Kharif/Bloomberg) &gt;&gt; https://t.co/J9i43tcz7x #news #tech
RT @codeyisfun: ITS MY FUCKING BIRTHDAY. 🔥

FEEL FREE TO SEND ME BITCOIN, PICS OF YOUR DOGS, OR DM SAYING HAPPY BIRTHDAY. 😋

Thank you. 

R…
RT @Airdropnotecom: #Bitcoin #Satoshi #Earn_Money #cryptocurrency #blockchain #Airdrop
New Airdrop #GlobaTalent 📢

GlobaTalent is airdroppi…
RT @CryptoFights: Check out some of the exclusive weapons only for CryptoFights! Be sure to follow us in the forums for a chance to win pri…
RT @BVBTC: "Nuanced Analysis of LocalBitcoins Data Suggests Bitcoin is Working as Satoshi Intended"

Thanks @MartyBent for writing about th…
START BITCOIN MINING       https://t.co/dPlJzmqpLA
RT @RoadtoRoota: I sat down with my old friend Sean at SGT Report to chat about the state of the world.
 
Here's the link:
 
This Feels Lik…
Russian nuclear

NHL - theScore 🏒 Report: Coyotes sign Kuemper to 2-year, $9M extension https://t.co/aqrqDPcyVy ►  https://t.co/0I4IIflkwI  √ https://t.co/mqUSNTmfVv
RT @Cointelegraph: Bitcoin’s scarcity is the main reason why it is going to hit $1 million in 2020, John McAfee says https://t.co/RjvernKopz
RT @eSportBettingPR: The Big Giveaway is on! 🎉🎉we are giving away 200 ESBC to 10 lucky winners.
1⃣Give like to this tweet
2⃣Follow @eSportB…
https://t.co/vUK0KWLdwh - Despite the controversies, Tether has become the most traded crypto and has ~18% higher monthly trading volume than that of Bitcoin, according to CoinMarketCap (Olga Kharif/Bloomberg) #tech #mobile
1,500,000 Bitcoin-Turbo-Koin (BTK) MINING-CONTRACT, 1.5 Million, Crypto Currency https://t.co/jDkmO6SI4B #bitcoin #crypto #antminer #bitmain
6x Antminer L3++ = 3400 MH/s – 24hr Scrypt Mining Contract – Special Rate https://t.co/TOYbRYLZMK #bitcoin #crypto #antminer #bitmain
RT @Rhythmtrader: Buying bitcoin is the most effective way to steal fr

@OTC_Bitcoin @KingThies @OTC_Bitcoiin Unrelated but what do you think of the head and shoulders on this one? Asking for a friend. https://t.co/Rqp3ENaDJp
RT @DiepSanh: - How Catherine Coley, CEO of Binance US, gained the experience in building liquidity for XRP at Ripple

- XRP was under-appr…
RT @NeerajKA: How would you navigate a world where an authoritarian government controls the payment system?  https://t.co/jQwigRSrSs
RT @Xbond49: The world’s most-used #Cryptocurrency isn’t #bitcoin https://t.co/yLReJrOOsf by @crypto 

#trading #stablecoins #bitcoin #Teth…
The fact that Bitcoin is not a true currency is why it solves the issue of money being a zero sum game. https://t.co/wumZ4QKmSq
RT @andre_spicer: Remember how cryptocurrencies were supposed to be ultra efficient? New study finds bitcoin generates a welfare loss about…
RT @BlockchainBizMg: The US Government &amp; Federal Reserve has printed more than 63.5 Billion dollars in the last two weeks. Funds have been…
RT @bquittem: bitc

In [5]:
import nltk

In [6]:
from nltk.tag import UnigramTagger

In [7]:
from nltk.corpus import treebank

In [8]:
train_sents = treebank.tagged_sents()[:3000]

In [9]:
tagger = UnigramTagger(train_sents)

In [10]:
from nltk.corpus import sentiwordnet as swn

In [13]:
def EtudierEmotionTexte(url) :
    tes = open(f"{url}",encoding='utf-8' )
    text = tes.read()
    text = text.replace('#','')
    text = text.split(' ')
    result_type_text = tagger.tag(text)
    resul = []
    for i in range(len(result_type_text)) :
        if result_type_text[i][1] in ["JJ", "JJR", "JJS", "RBR", "RBS"] :
            resul.append(result_type_text[i][0])
    tes = []
    for i in resul :
        tes.append(swn.senti_synsets(i))
    keepFirstOne = []
    for i in tes :
        keepInMemory = i
        listI = list(i)
        if(len(listI) != 0) :
            keepFirstOne.append(listI[0])
        i = keepInMemory
    sommePos = 0
    sommeNeg = 0
    for i in keepFirstOne :
        if(i.pos_score()>i.neg_score()) :
            sommePos = sommePos + i.pos_score()
        sommeNeg = sommeNeg + i.neg_score()
    print(f"{sommePos} {sommeNeg}")
    if sommePos > sommeNeg : 
        return "Pos"
    else :
        return "Neg"

In [11]:
def EtudierEmotionTexte(url) :
    tes = open(f"{url}",encoding='utf-8' )
    text = tes.read()
    text = text.replace('.','')
    text = text.replace('!','')
    text = text.replace(',','')
    text = text.replace('?','')
    text = text.replace('#','')
    text = text.split(' ')
    result_type_text = tagger.tag(text)
    resul = []
    for i in range(len(result_type_text)) :
            resul.append(result_type_text[i][0])
    tes = []
    for i in resul :
        tes.append(swn.senti_synsets(i))
    keepFirstOne = []
    for i in tes :
        keepInMemory = i
        listI = list(i)
        if(len(listI) != 0) :
            keepFirstOne.append(listI[0])
        i = keepInMemory
    sommePos = 0
    sommeNeg = 0
    for i in keepFirstOne :
        if(i.pos_score()>i.neg_score()) :
            sommePos = sommePos + i.pos_score()
        sommeNeg = sommeNeg + i.neg_score()
    print(f"{sommePos} {sommeNeg}")
    if sommePos > sommeNeg : 
        return "Pos"
    else :
        return "Neg"

In [14]:
print(EtudierEmotionTexte("test2.txt"))

458.0 355.5
Pos


Le 27/09/2019
## 335.125 120.75 seulement adverbe et adjectif
## 3422.75 1677.625 tout le texte
## 3684.125 1879.375 Suppression de la ponctuation

Le 30/09/2019 
# 3071.25 2482.081999999997 Suppression de la ponctuation
# 458 355.5 Tout le texte

# Version test du package de Raphaël

In [2]:
from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer

In [3]:
analyser = SentimentIntensityAnalyzer()

In [13]:
import pandas as pd


In [ ]:
tweetScore = pd.DataFrame(sentiment)
for tweetText in tweetCleanArray:
    score = analyser.polarity_scores(tweetText)
    sentiment.append(score)

In [57]:
def EtudierEmotionTexteR(url) :
    sentiment = []
    tes = open(f"{url}",encoding='utf-8' )
    text = tes.read()
    text = text.replace('.','')
    text = text.replace('!','')
    text = text.replace(',','')
    text = text.replace('?','')
    text = text.replace('#','')
    text = text.split(' ')
    #result_type_text = tagger.tag(text)
    #resul = []
    #for i in range(len(result_type_text)) :
    #    if result_type_text[i][1] in ["JJ", "JJR", "JJS", "RBR", "RBS"] :
    #        resul.append(result_type_text[i][0])
    #tes = []
    for tweet in text :
        score = analyser.polarity_scores(tweet)
        sentiment.append(score)
    tweetScore = pd.DataFrame(sentiment)
    print(tweetScore.pos.sum())
    print(tweetScore.neg.sum())
    print(tweetScore.mean())

In [61]:
EtudierEmotionTexteR("test3.txt")

9964.012999999997
3340.5380000000005
compound    0.010518
neg         0.014445
neu         0.883240
pos         0.043086
dtype: float64


Le 27/09 :
Number Pos : 5930.544999999998
Number Neg : 1695.179
compound    0.013088
neg         0.012271
neu         0.856211
pos         0.042930

Le 30/09 :
8432.651000000002
2028.686
compound    0.017799
neg         0.013978
neu         0.873808
pos         0.058104

Le 02/10
Number Pos : 9964.012999999997
Number Neg : 3340.5380000000005
compound    0.010518
neg         0.014445
neu         0.883240
pos         0.043086